cdc/sink/ddlsink/mysql/async_ddl_test.go

```diff
@@ -33,18 +32,13 @@ import (
 )
 
 func TestWaitAsynExecDone(t *testing.T) {
-	var dbIndex int32 = 0
-	GetDBConnImpl = func(ctx context.Context, dsnStr string) (*sql.DB, error) {
-		defer func() {
-			atomic.AddInt32(&dbIndex, 1)
-		}()
-		if atomic.LoadInt32(&dbIndex) == 0 {
-			// test db
-			db, err := pmysql.MockTestDB()
-			require.Nil(t, err)
-			return db, nil
-		}
-		// normal db
+	dbConnFactory := pmysql.DBConnectionFactoryForTest{}
+	dbConnFactory.SetTemporaryConnectionFactory(func(ctx context.Context, dsnStr string) (*sql.DB, error) {
```

Comment:
The implementation of SetTemporaryConnectionFactory is the same at all calls, maybe it should be handled during construction to reduce redundant code.

Reply:
Thank you for your suggestion! That's a good idea; moving the implementation of SetTemporaryConnectionFactory to a new constructor is a good way to reduce this redundant code. I’ll make the necessary changes and update the PR shortly.